tower/examples/server.rs

```diff
@@ -0,0 +1,81 @@
+extern crate futures;
+extern crate hyper;
+extern crate tokio_tcp;
+extern crate tower;
+extern crate tower_hyper;
+extern crate tower_in_flight_limit;
+extern crate tower_service;
+
+use futures::{future, Future, Poll, Stream};
+use hyper::{Body, Request, Response};
+use tokio_tcp::TcpListener;
+use tower::builder::ServiceBuilder;
+use tower_hyper::body::LiftBody;
+use tower_hyper::server::Server;
+// use tower_in_flight_limit::InFlightLimitLayer;
+use tower_service::Service;
+
+fn main() {
+    hyper::rt::run(future::lazy(|| {
+        let addr = "127.0.0.1:3000".parse().unwrap();
+        let bind = TcpListener::bind(&addr).expect("bind");
+
+        println!("Listening on http://{}", addr);
+
+        let maker = ServiceBuilder::new(MakeSvc)
+            // TODO(lucio): figure out why this is causing sized issues
+            // .add(InFlightLimitLayer::new(5))
```

Comment:
Having a sized issue here when I uncomment this line.

![image](https://user-images.githubusercontent.com/5758045/54443875-f9aa0580-4717-11e9-8f01-1f9114f5ccd7.png)

Reply:
It's this line here: https://github.com/tower-rs/tower-hyper/blob/master/src/server/mod.rs#L24

It wants the error type to implement `Error`, and `Box<E>` only implements `Error` if it is `Sized` (not a trait object).

Reply:
Okay, so I tried to change this to `Into<Box<dyn Error>>` and got the same thing :/

Reply:
Oh here too :D https://github.com/tower-rs/tower-hyper/blob/master/src/server/mod.rs#L81

Reply:
Fixed this one!